File: style/style_user_testing/binary_search_bad.py
# Python code to implement iterative Binary Search. It returns location of z in given array arr if present, else returns -1 
def binarySearch(arr, x, y, z): 
  
    while x <= y: 
  
        mid = x + (y - x)/2; 
        # print mid

        if arr[mid] == z: 
            return mid 
  
        elif arr[mid] < z: 
            x = mid + 1
  
        else: 
            y = mid - 1
      
    # If we reach here, then the element 
    # was not present 
    return -1

File: style/style_user_testing/binary_search_bad.c
// C program to implement iterative Binary Search 
#include <stdio.h> 
  
// A iterative binary search function. It returns location of z in given array arr[l..r] if present, otherwise -1 
int binarySearch(int arr[], int x, int y, int z) 
{ 
    while (x <= y) { 
          int m = x + (y - x) / 2; 
  
            //printf("%d\n", m);
          //printf("%d, %d", x, y);
        if (arr[m] == z) 
            return m; 
  
            if (arr[m] < z) 
                x = m + 1; 
  
            else
              y = m - 1; 
    } 
    return -1; 
} 